Makefile

```diff
@@ -31,20 +31,23 @@ include $(DMLC_CORE)/make/dmlc.mk
 # include the plugins
 include $(XGB_PLUGINS)
 
-# use customized config file
+# set compiler defaults for OSX versus *nix
+# let people override either
+OS := $(shell uname)
+ifeq ($(OS), Darwin)
 ifndef CC
-export CC  = $(if $(shell which gcc-6),gcc-6,gcc)
+export CC = $(if $(shell which clang), clang, gcc)
 endif
 ifndef CXX
-export CXX = $(if $(shell which g++-6),g++-6,g++)
+export CXX = $(if $(shell which clang++), clang, g++)
+endif
+else
```

Comment:
Sorry this `else` is for Linux? Might be worth adding a short comment to make this clear.